packages/protocol/contracts/governance/Validators.sol

```diff
@@ -258,27 +260,32 @@ contract Validators is
 
   /**
    * @notice Registers a validator unaffiliated with any validator group.
-   * @param publicKeysData Comprised of three tightly-packed elements:
-   *    - publicKey - The public key that the validator is using for consensus, should match
-   *      msg.sender. 64 bytes.
-   *    - blsPublicKey - The BLS public key that the validator is using for consensus, should pass
-   *      proof of possession. 48 bytes.
-   *    - blsPoP - The BLS public key proof of possession. 96 bytes.
+   * @param ecdsaKey The ECDSA public key that the validator is using for consensus, should match
+   *   the validator signer. 64 bytes.
+   * @param blsKey The BLS public key that the validator is using for consensus, should pass proof
+   *   of possession. 48 bytes.
+   * @param blsPop The BLS public key proof-of-possession, which consists of a signature on the
+   *   account address. 96 bytes.
    * @return True upon success.
    * @dev Fails if the account is already a validator or validator group.
    * @dev Fails if the account does not have sufficient Locked Gold.
    */
-  function registerValidator(bytes calldata publicKeysData) external nonReentrant returns (bool) {
-    address account = getAccounts().activeValidationSignerToAccount(msg.sender);
+  function registerValidator(bytes calldata ecdsaKey, bytes calldata blsKey, bytes calldata blsPop)
+    external
+    nonReentrant
+    returns (bool)
+  {
+    address account = getAccounts().activeValidatorSignerToAccount(msg.sender);
     require(!isValidator(account) && !isValidatorGroup(account));
     uint256 lockedGoldBalance = getLockedGold().getAccountTotalLockedGold(account);
     require(lockedGoldBalance >= validatorLockedGoldRequirements.value);
     Validator storage validator = validators[account];
-    _updatePublicKeysData(validator, publicKeysData);
-    validator.publicKeysData = publicKeysData;
+    address signer = getAccounts().getValidatorSigner(account);
```

Comment:
Is this not always equal to `msg.sender` by virtue of having obtained `account` via `activeValidatorSignerToAccount`?

Reply:
Not necessarily, `msg.sender` could be the account address

Reply:
I don't understand. Is it not true that `x == getValidatorSigner(activeValidatorSignerToAccount(x))`?

Reply:
That will be true if `x` is an authorized signer, but not if `x` is an account address.

Reply:
Oh I see now. Thanks for explaining that.